merino-adm/src/remote_settings/reqwest_client.rs

```diff
@@ -20,10 +20,13 @@ pub struct ReqwestClient {
 
 impl ReqwestClient {
     /// Instantiate a new Reqwest client to perform HTTP requests.
-    pub fn new() -> ReqwestClient {
-        Self {
-            reqwest_client: reqwest::Client::new(),
-        }
+    pub fn try_new() -> Result<ReqwestClient, Error> {
+        let reqwest_client = reqwest::Client::builder()
+            // Disable the connection pool to avoid the IncompleteMessage errors.
+            // See #259 for more details.
+            .pool_max_idle_per_host(0)
```

Comment:
I'm wondering why this wasn't happening with the other implementation: we were not setting the max idle per host there as well. Any idea?

Aside from that, I don't think there's a big problem in disabling pooling.